catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,680 @@
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ServiceBindingType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+
+public class SourceConfigurationHandler implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SourceConfigurationHandler.class);
+
+    private static final String METACARD_PROPERTY = "ddf.catalog.event.metacard";
+
+    private static final String BINDING_TYPE = "bindingType";
+
+    private static final String DISABLED_CONFIGURATION_SUFFIX = "_disabled";
+
+    private static final String ID = "id";
+
+    private static final String SHORTNAME = "shortname";
+
+    private static final String CONFIGURATION_FILTER =
+            "(&(%s=%s)(|(service.factoryPid=*source*)(service.factoryPid=*Source*)(service.factoryPid=*service*)(service.factoryPid=*Service*)))";
+
+    private static final int SHUTDOWN_TIMEOUT_SECONDS = 60;
+
+    private ConfigurationAdmin configurationAdmin;
+
+    private FederationAdminService federationAdminService;
+
+    private MetaTypeService metaTypeService;
+
+    private Parser parser;
+
+    private ParserConfigurator unmarshalConfigurator;
+
+    private ExecutorService executor;
+
+    private String urlBindingName;
+
+    private Map<String, String> bindingTypeToFactoryPidMap = new HashMap<>();
+
+    private List<String> sourceActivationPriorityOrder = new ArrayList<>();
+
+    private Boolean activateConfigurations;
+
+    private Boolean preserveActiveConfigurations;
+
+    private Boolean cleanUpOnDelete;
+
+    public SourceConfigurationHandler(FederationAdminService federationAdminService,
+            ExecutorService executor) {
+        this.federationAdminService = federationAdminService;
+        this.executor = executor;
+    }
+
+    public void destroy() {
+        executor.shutdown();
+        try {
+            if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                executor.shutdownNow();
+                if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                    LOGGER.error("Thread pool didn't terminate");
+                }
+            }
+        } catch (InterruptedException e) {
+            executor.shutdownNow();
+        }
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null || !mcard.getTags()
+                .contains(RegistryConstants.REGISTRY_TAG)) {
+            return;
+        }
+
+        executor.execute(() -> {
+            if (event.getTopic()
+                    .equals("ddf/catalog/event/CREATED")) {
+                processCreate(mcard);
+            } else if (event.getTopic()
+                    .equals("ddf/catalog/event/UPDATED")) {
+                processUpdate(mcard);
+            } else if (event.getTopic()
+                    .equals("ddf/catalog/event/DELETED")) {
```

Comment:
should these three strings be constants? or even split up into 4 constants, eg `private static final String BASE_TOPIC = "ddf/catalog/event/"` and then one for each of `"CREATED"`, `"UPDATED"` and `"DELETED"`?